### PR TITLE
codegen: mixed-args, splat, and typed block params for instance_exec trampolines

### PR DESCRIPTION
**Pull request by ryanseys**
Extends the pre-existing proc-arg `instance_exec` trampoline inliner (`def m(*args, &b); instance_exec(*args, &b); end`) beyond fixed-arity local forwarding, and fixes block-param type resolution in the spliced body. 

## What's here
- `cls_method_trailing_proc_param_name` accepts a trailing `&blk` typed `poly` (forwarding-only) as well as `proc`, and allows rest params before it.
- `is_instance_exec_trampoline` accepts each trampoline-body positional arg that is a forwarded local, a literal, the receiver's ivar, or a sole `SplatNode` of the rest param.
- `compile_instance_exec_inlined_stmt` evaluates each arg per-kind at the call site; splat trampolines fan out so different call sites can pass different arg counts.
- `find_var_type` inside the splice resolves block-param names to their typed slot (declared under both the per-call-site suffixed name and the original name, bracketed by `push_scope` / `pop_scope`).

## Tests
`instance_exec_spliced_types` (string/symbol/int params + outer shadow), `instance_exec_mixed_args` (local + ivar + literal forwarding), `instance_exec_splat` (per-call-site arity fan-out).

## Verification
`make bootstrap` fixpoint passes; `make retest` green (864 tests).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/instance_exec_splat.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Three args at one call site, two args at another -- the splat
+# trampoline fans out each call's args to match its block params.
+b.with_all(1, 2, 3) { |a, b, c| add(a + b + c) }      # +6
+b.with_all(10, 20)  { |x, y|    add(x + y) }          # +30
+b.with_all(100)     { |n|       add(n) }              # +100
+puts b.total                                           # 136
+
+# Same trampoline on a second instance, zero args edge case.
+b2.with_all(5, 5, 5, 5) { |a, b, c, d| add(a + b + c + d) }   # +20
```
[the remark this answers is not in the document]
Comment (ryanseys) [reply test/instance_exec_splat.rb]: resolved in 6fe8151a

File: test/instance_exec_spliced_types.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  def add(n); @s = @s + n; end
+  def total; @s; end
+end
+
+class BuilderPlus < Builder
+end
+
+b = Builder.new
+t.fire("hi", :sym, 7) { |s, y, n| b.add(n) }   # `b` here is the outer Builder
+t.fire("hi", :sym, 7) { |s, y, n| b.add(n) }
```
[the remark this answers is not in the document]
Comment (ryanseys) [reply test/instance_exec_spliced_types.rb:50]: resolved in 6fe8151a

**Comment by matz** [comment]
Implemented on master (commit 1b01917). A trampoline whose body passes mixed args — `instance_exec(x, @base, 7, &b)` (method param + ivar + literal) — now binds the caller block's params to the trampoline body's args (with the caller's argument substituted for a trampoline-param read), instead of to the caller's args directly. The splat-forwarding and spliced-types forms already worked. Ported from your test files.
